src/widgets/search-box/search-box.tsx

```diff
@@ -28,15 +28,15 @@ export type SearchBoxTemplates = Partial<{
   /**
    * Template used for displaying the submit button. Can accept a function or a Hogan string.
    */
-  submit: Template;
+  submit: Template<{ cssClasses: SearchBoxComponentCSSClasses }>;
```

Comment:
would it save any errors if Template required the generic? are there use cases for generic-less template left?